### ISSUE TITLE
SPMC 16.2.0 - 23.976 FPS Video Stutters Every Few Minutes

### ISSUE THREAD

**Issue by Alamei** [koying/SPMC#203]
Since I installed the latest version of SPMC (16.2.0) on my NVIDIA Shield, I've noticed a weird glitch when playing back 23.976 FPS video: Approximately every three and a half minutes, the video stutters for around 10 seconds. I've tested several source source files with varying video resolutions, and it occurs in every one of them. The issue doesn't occur in the March 8th pass-through test build of Kodi when running the same files with the same settings, nor did I see it with older versions of SPMC 16.x. (**EDIT**: It appears I was mistaken about this fact, and was just assuming that the glitches were source-related when I was on earlier 16.x builds; see comment from April 21. The Kodi build doesn't present the issue, but all of the SPMC versions I tested do.)

Most of my settings are left at the defaults, with the following exceptions:
- Alter refresh rate is set to: On Start/Stop
- Delay on refresh change: 3 seconds
- Sync playback to display is: Off
- Vertical blank sync is: Off
- Audio pass-through is: Enabled

I tried generating a debug log, but it created a 28MB log file for just eight minutes of video playback (enough to allow the error to occur twice), which is too large to upload to any of the sites. I'm guessing that's due to the individual component debug logging for video and audio; do you need those individual component logs, or can I run a debug log without them and upload that?

**Comment by dubastik** [comment]
Hi alamei i have a question because i have somes freezes on  vidéos with spmc, you are in full android or with the rom stock ?

**Comment by Alamei** [comment]
I'm not sure I understand your question. My Shield is running the 3.0 update (corresponding to Android 6.0 Marshmallow), and I haven't rooted the device or taken any other action to modify the base ROM, so I would consider my system "stock" for all intents and purposes.

**Comment by dubastik** [comment]
Excuse me for my english,i speak french,ok thank's for your answer !!

**Comment by jirkav77** [comment]
The same on Amlogic S805 device, Only 23.976 fps video causes the problem, the same problem occurs also with bundled Kodi 14.1 Helix. The TV set is correctly switched to 24p display mode. The problem is not present with 25 fps videos (and 50p display mode). It seems like something wrong with syncing 23.976 -> 24 fps. It doesn't look like the "one frame added each 41 seconds".

**Comment by Alamei** [comment]
I this you might be experiencing a different issue, since the issue I've noted in this topic does not occur in the Kodi builds I've tried. (I'm temporarily using the March 8th Kodi pass-through test build as my primary video player while I try to figure out what's causing the issue with SPMC.)

Have you timed the stutter you're seeing to check if (a) it occurs at roughly three-and-a-half minute intervals and (b) it lasts approximately 10 seconds each time it occurs?

**Comment by jirkav77** [comment]
You're right, Alamei, my problem is different - video stutters each approx 41 secs (incorrect sync 23.976 fps to 24p display). It may be problem of amlogic driver.

**Comment by Alamei** [comment]
My Shield just updated to 3.1, and I pushed the new version of SPMC (16.2.1) a couple of days ago, so I decided to re-run tests for this issue. Unfortunately, it looks like the stutter is still present for all files played back in SPMC 16.2.1 on Shield 3.1. The Kodi passthrough test build still plays back the files without stutter on 3.1, as well.

Here is a debug log (sans any video/audio component logs, which, as mentioned previously, make the verbose debug log absolutely massive): [SPMC 16.2.1 Log](http://www.xbmclogs.com/p3jfzd75d)

**Comment by Alamei** [comment]
In the interest of thorough testing, I uninstalled the current version of SPMC and loaded the old 16.1.2 version back onto my Shield. It appears that I was mistaken about the issue not occurring in the older versions. When I watched a couple of test videos closely, I saw that the issue was still presenting approximately every 3.5 minutes, although it seemed less noticeable (perhaps slightly less pronounced).

In any case, it appears that the issue impacts all versions of SPMC, not just the recent 16.2.x builds. Is there any other testing I can do (on any version) to help troubleshoot?

**Comment by SamuelBolduc** [comment]
Just a heads up to let you know I have this exact same problem too, with the same settings enabled. If it can have anything to do with it, by AVR is a [Denon S510-BT](https://usa.denon.com/us/product/hometheater/receivers/avrs510bt). Other specs are pretty much standard: nVidia Shield on FW 3.1, SPMC 16.3.0

**Comment by Alamei** [comment]
Thanks for reporting in on the issue. I have my Shield routed through a Yamaha RX-V375, so it doesn't look like the bug is related to specific AVRs.

I'm very curious if this issue only impacts a minority of users, or if most users just don't notice the stutter or chalk it up to an issue with the source file/network/etc. Koying, could you chime in on whether this issue is reproducible on your hardware?

**Comment by Alamei** [comment]
I noticed that there was mention of 23.976 FPS fixes in the Shield 3.2 update, so I re-ran some tests with SPMC 16.3.0, and the issue appears to be resolved on my setup.

**EDIT:** It appears I spoke to soon. The issue is definitely improved, since it doesn't show up on every single 23.976 file every 3.5 minutes, but it does still occur sporadically.

**Comment by Dariusz30** [comment]
There is a bug with 23,976 on nVidia Shield 3.2 update - on my connection Epson eh tw 3600 - Pioneer vsx 924 ( 1080 p /24 p ) i still have stutter...even on Shield menu ...on 1080 p /60 Hz no problem ...

**Comment by Alamei** [comment]
The issue you're experiencing doesn't sound like it's related to this at all, Darius. You should post on the NVidia forums for troubleshooting any general Shield bugs.
